File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/like/service/ReadLIkeService.java
package io.dustin.apps.board.domain.like.service;

import io.dustin.apps.board.domain.like.model.Like;
import io.dustin.apps.board.domain.like.repository.LikeRepository;
import io.dustin.apps.common.code.BoardType;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import static io.dustin.apps.common.utils.OptionalUtils.getEntity;

@RequiredArgsConstructor
@Service
public class ReadLIkeService {

    private final LikeRepository likeRepository;

    @Transactional(readOnly = true)
    public Like getLike(Long userId, Long boardId, BoardType boardType) {
        return getEntity(this.likeRepository.findByUserIdAndBoardIdAndBoardType(userId, boardId, boardType), Like.class, "like not found");
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/like/service/ReadLIkeService.kt
package io.dustin.apps.board.domain.like.service

import io.dustin.apps.board.domain.like.model.Like
import io.dustin.apps.board.domain.like.repository.LikeRepository
import io.dustin.apps.common.code.BoardType
import io.dustin.apps.common.utils.notFoundEntity
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ReadLIkeService (

    private val likeRepository: LikeRepository

) {

    /**
     * 좋아요 객체 조회
     */
    @Transactional(readOnly = true)
    fun getLikeObj(userId: Long, boardId: Long, boardType: BoardType): Like? {
        return likeRepository.findByUsers(userId, boardId, boardType)

    }

    /**
     * 좋아요 객체 조회
     */
    @Transactional(readOnly = true)
    fun getLike(userId: Long, boardId: Long, boardType: BoardType): Like {
        val errorMessage = "해당 값: [$userId]로 조회된 정보가 없습니다."
        return likeRepository.findByUsers(userId, boardId, boardType) ?: notFoundEntity(errorMessage)

    }


}
